src/crimson/os/seastore/seastore.cc

```diff
@@ -1791,20 +1791,23 @@ SeaStore::Shard::_do_transaction_step(
       //TODO: use when_all_succeed after making onode tree
       //      support parallel extents loading
       return onode_manager->get_or_create_onode(*ctx.transaction, dest_oid
-      ).si_then([&onodes, &d_onodes, op](auto onodevec) {
-	auto dest_onode = onodevec.onode;
+      ).si_then([&onodes, &d_onodes, op,
+		 onodevec=std::move(onodevec)](auto d_onodevec) {
+	auto dest_onode = d_onodevec.onode;
 	assert(dest_onode);
 	auto &d_o = onodes[op->dest_oid];
 	assert(!d_o);
 	assert(!d_onodes[op->dest_oid]);
 	d_o = dest_onode;
 	d_onodes[op->dest_oid] = dest_onode;
```

Comment:
CC @xxhdx1985126 I was trying to recall why `d_onodes` was introduced.

Looking at https://github.com/ceph/ceph/pull/53234, it seems to me that `d_onodes` was originally introduced as "dirty_onodes" for `write_dirty()`, but the requirement is no longer necessary after the fix.

Somehow, `d_onodes` was parsed as `dest_onodes` afterwards (i.e. incomplete name leads to misunderstandings), and seems to me this is not reasonable. 

If I'm correct above, can we take the chance to drop `d_nodes` to simplify the code ?

Reply:
Ah, yes, it seems so, will remove it.